src/hotspot/share/oops/compressedKlass.cpp

```diff
@@ -68,7 +176,13 @@ void CompressedKlassPointers::initialize_for_given_encoding(address addr, size_t
   _base = requested_base;
   _shift = requested_shift;
 
-  DEBUG_ONLY(assert_is_valid_encoding(addr, len, _base, _shift);)
+  calc_lowest_highest_narrow_klass_id();
+
+#ifdef ASSERT
+  sanity_check_after_initialization();
+#endif
+
+  DEBUG_ONLY(sanity_check_after_initialization();)
```

Comment:
This is here twice.

Reply:
sanity_check_after_initialization is called from both initialization routines, but we only call either one or the other.